### PR TITLE
Update test code

### PR DESCRIPTION
**Pull request by beru**
Hi, I updated the test code for your PR (https://github.com/tiny-dnn/tiny-dnn/pull/747)
I hope this PR will be merged to your branch.

And I noticed some code are outdated compared to `tiny_dnn/master` branch, so maybe you should rebase your `serial_type` branch to origin master branch.

### CODE REVIEW

File: tiny_dnn/util/serialization_functions.h [hunk 1]
```diff
@@ -47,75 +47,6 @@ void arc(Archive &ar, cereal::NameValuePair<size_t> &&arg) {
   arg.value = arg2.value;
 }
 
-static serial_size_t to_serial(const size_t &src) {
```
Comment (edgarriba) [inline tiny_dnn/util/serialization_functions.h:50]: why would you remove this?

Comment (edgarriba) [reply tiny_dnn/util/serialization_functions.h:50]: and below

Comment (beru) [reply tiny_dnn/util/serialization_functions.h:50]: I confirmed they aren't actually needed. Only `serial_size_t` <=> `size_t` conversion wrapped in `cereal::NameValuePair` type is enough. 

With other arc functions and this function
```cpp
template <class Archive, typename T>
void serialize(Archive &ar, tiny_dnn::index3d<T> &idx) {
  arc(ar, make_nvp("width", idx.width_), make_nvp("height", idx.height_),
      make_nvp("depth", idx.depth_));
}
```
cereal library can do the job (ah.... horrible code generation).

**Comment by edgarriba** [comment]
oh, really? nice!

**Pull request merged**